helix-term/src/commands.rs

```diff
@@ -1010,6 +1023,52 @@ mod cmd {
         doc.later(view.id, uk)
     }
 
+    fn write_quit(editor: &mut Editor, args: &[&str], event: PromptEvent) {
+        write(editor, args, event.clone());
+        quit(editor, &[], event);
+    }
+
+    fn force_write_quit(editor: &mut Editor, args: &[&str], event: PromptEvent) {
+        write(editor, args, event.clone());
+        force_quit(editor, &[], event);
+    }
+
+    fn _write_all(editor: &mut Editor, args: &[&str], event: PromptEvent, quit: bool, force: bool) {
+        let ids = editor
+            .tree
+            .views()
+            .map(|(view, _)| (view.id, view.doc))
+            .collect::<Vec<_>>();
+
+        for (view_id, doc_id) in ids {
```

Comment:
I think you should be just looping over `editor.documents` to save, then loop over views to close?

If there's buffers open in the background and not visible in a split, this will fail to save those (and then force closing will lose the changes!)

Reply:
The problem when iterating over `editor.documents` is that I can't `set_error` on editor because it's already mutably borrowed.

Reply:
Maybe store the message then set it once out of the loop? I'd like to find a better interface to accessing documents too, the `.current()` call feels like a bit of a hack.

Reply:
I ended up doing what you suggested, I'm still unsure how to handle multiple errors. For now they are stored in a single string separated by a newline and later displayed at once.